### ISSUE TITLE
Voice System

### ISSUE THREAD

**Issue by laqieer** [laqieer/FEHRG#2]
4 voices (map / damage / status / attack) per character to save space.

**Comment by laqieer** [comment]
The sound quality is improved by changing sampling frequency from 5734 to 10512. The DPCM compression reduces the increase of ROM size. The size only increases 1.35MB.

SNR-frequency relation:
![image](https://user-images.githubusercontent.com/8841957/136909638-dd8ec448-599a-4e6e-8c82-73841d8a9eba.png)
In theory, lower sampling frequency => less sampling points => greater distance between points => more quadratuer error => lower SNR

SNR distribution when frequency = 10512:
![image](https://user-images.githubusercontent.com/8841957/136909869-12390ea2-0698-4736-931d-c0af3a238021.png)
In theory, when digitizing sound, it has been found that you must use a sampling rate of twice the highest frequency of sound that you want to digitize in order to accurately reproduce the sound. This is called Nyquist’s theorem. The human speaking voice typically has frequencies of up to 4000 Hz, so a sampling rate of 8000 Hz is sufficient to sample it with an acceptable quality. This is why a sampling frequency of 8 kHz is often found within codecs used for VoIP. GBA accepts sampling rates (Hz): 5734, 7884, 10512, 13379, 15768, 18157, 21024, 26758, 31536, 36314, 40137, 42048. Therefore we choose 10512 (>8k) here.

SNR distribution when frequency = 5734:
![image](https://user-images.githubusercontent.com/8841957/136910039-32058ba5-b823-4813-a7f8-6a3300e5e9d9.png)
The quality is too poor for lowest frequency 5734, though it saves ROM space.

ROM size (KB) with min-SNR limit when frequency = 5734:
![image](https://user-images.githubusercontent.com/8841957/136911830-91ed9de5-86c4-4278-beb6-cfcdc4c13fad.png)
The size increases significantly after SNR = 5, so we set min SNR for compression to 5.

In conclusion, I think the current default parameters is good.
```
FREQ = 10512
SNR = 5
```

**Comment by laqieer** [comment]
Correction: https://github.com/laqieer/se2m4a/commit/8082dc1542d092b514bdaf569eb410aa78519570

![SNR-Freq](https://user-images.githubusercontent.com/8841957/136944439-4a265bb4-3d00-43a8-8c8e-c904d1a9ba77.png)
test script: https://github.com/laqieer/se2m4a/blob/master/tool/test_frequency_and_SNR.py
test data:
```
test_data/VOICE_Abel_The_Panther_ATTACK_1.wav
test_data/VOICE_Abel_The_Panther_ATTACK_1_jp.wav
test_data/VOICE_Abel_The_Panther_ATTACK_2.wav
test_data/VOICE_Abel_The_Panther_ATTACK_2_jp.wav
test_data/VOICE_Abel_The_Panther_DAMAGE_1.wav
test_data/VOICE_Abel_The_Panther_DAMAGE_1_jp.wav
test_data/VOICE_Abel_The_Panther_DAMAGE_2.wav
test_data/VOICE_Abel_The_Panther_DAMAGE_2_jp.wav
test_data/VOICE_Abel_The_Panther_DEAD_1.wav
test_data/VOICE_Abel_The_Panther_DEAD_1_jp.wav
test_data/VOICE_Abel_The_Panther_MAP_1.wav
test_data/VOICE_Abel_The_Panther_MAP_1_jp.wav
test_data/VOICE_Abel_The_Panther_MAP_2.wav
test_data/VOICE_Abel_The_Panther_MAP_2_jp.wav
test_data/VOICE_Abel_The_Panther_MAP_3.wav
test_data/VOICE_Abel_The_Panther_MAP_3_jp.wav
test_data/VOICE_Abel_The_Panther_SKILL_1.wav
test_data/VOICE_Abel_The_Panther_SKILL_1_jp.wav
test_data/VOICE_Abel_The_Panther_SKILL_2.wav
test_data/VOICE_Abel_The_Panther_SKILL_2_jp.wav
test_data/VOICE_Abel_The_Panther_SKILL_3.wav
test_data/VOICE_Abel_The_Panther_SKILL_3_jp.wav
test_data/VOICE_Abel_The_Panther_SKILL_4.wav
test_data/VOICE_Abel_The_Panther_SKILL_4_jp.wav
test_data/VOICE_Abel_The_Panther_STATUS_1.wav
test_data/VOICE_Abel_The_Panther_STATUS_1_jp.wav
test_data/VOICE_Abel_The_Panther_STATUS_2.wav
test_data/VOICE_Abel_The_Panther_STATUS_2_jp.wav
test_data/VOICE_Abel_The_Panther_STATUS_3.wav
test_data/VOICE_Abel_The_Panther_STATUS_3_jp.wav
test_data/VOICE_Abel_The_Panther_STATUS_4.wav
test_data/VOICE_Abel_The_Panther_STATUS_4_jp.wav
```
[A related research](https://abrightersun.de/blog/dpcm-analyzed.html)

freq = 5734:
![Figure_1](https://user-images.githubusercontent.com/8841957/136944692-cd2f09a7-9132-44ec-bfaa-2031c9da3e2d.png)
`26426072 10 12 18:02 FEHRG.gba`

freq = 7884:
![Figure_2](https://user-images.githubusercontent.com/8841957/136944720-f314cc29-a828-415f-ab64-0a98b78c8048.png)
`27225664 10 12 18:41 FEHRG.gba`

freq = 10512:
![Figure_3](https://user-images.githubusercontent.com/8841957/136944858-c6d9b99a-0722-4743-9f85-f99425c32380.png)
`28203312 10 12 18:21 FEHRG.gba`

freq = 10512 & SNR >= 15:
![Figure_4](https://user-images.githubusercontent.com/8841957/136945042-54cb4e8b-7a98-4461-bbb4-38a2eee8505f.png)
`28222992 10 12 18:48 FEHRG.gba`
